cli/command/service/list.go

```diff
@@ -95,34 +95,51 @@ func PrintNotQuiet(out io.Writer, services []swarm.Service, nodes []swarm.Node,
 	}
 
 	running := map[string]int{}
+
+	// first key is service id, the value of first key is a map
+	// In the second map, key is node ID, value is task number of this service on this node.
+	tasksPerService := map[string]map[string]int{}
```

Comment:
You can implement the same thing with a struct-keyed map but I don't see this being accessed by taskID, so I don't quite understand what is being attempted here.

This should be called `tasksByService` or `taskCountByService`?